include/ginkgo/core/matrix/csr.hpp

```diff
@@ -771,7 +826,7 @@ class Csr : public EnableLinOp<Csr<ValueType, IndexType>>,
           col_idxs_(exec, num_nonzeros),
           row_ptrs_(exec, size[0] + 1),
           srow_(exec, strategy->clac_size(num_nonzeros)),
-          strategy_(std::move(strategy))
+          strategy_(std::move(strategy->copy()))
```

Comment:
I believe we no longer need to move it after the copy.
```suggestion
          strategy_(strategy->copy())
```